src/v/storage/index_state.cc

```diff
@@ -304,6 +304,14 @@ bool index_state::maybe_index(
     // always saving the first batch simplifies a lot of book keeping
     if ((accumulator >= step && user_data) || is_empty) {
         auto offset_delta = batch_base_offset() - base_offset();
+        // on sparse compacted topic indexes, there could be a case where the space
+        // between valid keys is greater than 4 bytes.
```

Comment:
nit: GB?

Reply:
no, meant more than can be represented in 32 bits.